detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ObjectLiteralToLambdaSpec.kt

```diff
@@ -0,0 +1,509 @@
+                            }
+                        }
+                    }
+                }
+                """
+                subject.compileAndLintWithContext(env, code)
+                    .assert()
+                    .hasSize(1)
+                    .hasSourceLocations(SourceLocation(6, 9))
+            }
+
+            it("is generic") {
+                val code = """
+                fun interface Sam<T> {
+                    fun foo(): T
+                }
+                val a = object : Sam<Int> {
+                    override fun foo(): Int {
+                        return 1
+                    }
+                }   
+                """
+                subject.compileAndLintWithContext(env, code)
+                    .assert()
+                    .hasSize(1)
+                    .hasSourceLocations(SourceLocation(4, 9))
+            }
+
+            it("has other default method") {
+                val code = """
+                fun interface Sam {
+                    fun foo()
+                    fun bar() {}
+                }
+                val a = object : Sam {
+                    override fun foo() {
+                    }
+                }   
+                """
+                subject.compileAndLintWithContext(env, code)
+                    .assert()
+                    .hasSize(1)
+                    .hasSourceLocations(SourceLocation(5, 9))
+            }
+
+            it("nested declaration") {
+                val code = """
+                interface First {
+                    fun foo()
+                }
+                fun interface Second: First
+
+                fun bar() {
+                    object : Second {
+                        override fun foo(){
+                        }
+                    }
+                }
+                """
+                subject.compileAndLintWithContext(env, code)
+                    .assert()
+                    .hasSize(1)
+                    .hasSourceLocations(SourceLocation(7, 5))
+            }
+
+            it("expression body syntax") {
+                val code = """
+                fun interface Sam {
+                    fun foo(): Int
+                }
+                val a = object : Sam {
+                    override fun foo() = 3
+                }
+                """
+                subject.compileAndLintWithContext(env, code)
+                    .assert()
+                    .hasSize(1)
+                    .hasSourceLocations(SourceLocation(4, 9))
+            }
+        }
+
+        context("is not correct implement") {
+            it("without context") {
+                val code = """
+                fun interface Sam {
+                    fun foo()
+                }
+                val a = object : Sam {
+                    override fun foo() {
+                    }
+                }
+                """
+                subject.compileAndLintWithContext(env, code).assert().isEmpty()
```

Comment:
```suggestion
                subject.compileAndLint(code).assert().isEmpty()
```

My bad, I didn't realize that this test is for non-typeresolution.